src/ast/query.rs

```diff
@@ -62,8 +62,18 @@ impl fmt::Display for Query {
             write!(f, "{with} ")?;
         }
         write!(f, "{}", self.body)?;
-        if !self.order_by.is_empty() {
-            write!(f, " ORDER BY {}", display_comma_separated(&self.order_by))?;
+        if !self.order_by.exprs.is_empty() {
+            write!(
+                f,
+                " ORDER BY {}",
+                display_comma_separated(&self.order_by.exprs)
+            )?;
+            if let Some(ref interpolate) = self.order_by.interpolate {
```

Comment:
I was conflicted about this - we're only going to write out `INTERPOLATE` if there are expressions, which makes sense (e.g. you can't have a free-standing `INTERPOLATE (...)` since it requires an `ORDER BY` but I suppose this _could_ be confusing since you can create:

```rust
OrderBy {
  exprs: vec![],
  interpolate: Some(Interpolate { ... }),
}
```

but you won't see `INTERPOLATE` written.

I don't know why anyone would do this as it's never valid so it seems OK since it means we will still have a valid query, instead of the alternative, which will always produce an invalid query.

Reply:
With this change, it would be possible to craft:

```rust
Some(OrderBy {
    exprs: vec![],
    interpolate: Interpolate { ... }
})
```

which would display:

```sql
ORDER BY INTERPOLATE
```

which gets interpreted as `INTERPOLATE` being a field name.

I'm not entirely sure that's preferable - it feels like every dialect is going to require _some_ identifier and then anything else added on. I'm not sure how much we can future-proof this for dialects that would accept an `ORDER BY` without identifiers but other keywords after. Maybe this falls under "user error" since it's possible to craft a bunch of scenarios that aren't actually valid and we entrust developers to understand how to use each AST node correctly.

---

I've pushed an update with this change in [44deb69](https://github.com/sqlparser-rs/sqlparser-rs/pull/1330/commits/44deb69db2ddb4c9448e9417f7e98fcd3056aa90), but can always revert it if we think it's problematic.

Reply:
I imainge that's fine as long as the parser itself doesn't wrongly accept SQL that builds up an invalid struct. in that, a user on the other hand can manually build up that struct however they see fit.